Tests/SentryProfilerTests/SentryProfilerSwiftTests.swift

```diff
@@ -449,18 +507,59 @@ private extension SentryProfilerSwiftTests {
-        try assertMetricValue(measurements: measurements, key: kSentryProfilerSerializationKeyFrozenFrameRenders, numberOfReadings: fixture.mockFrozenFramesPerBatch * metricsBatches, expectedValue: dummyValue, transaction: transaction)
-
-        // need to add 1 frame rate reading because there is always an initial one for the frame rate when the frame tracker starts
-        let expectedFrameRateReadings = 1 + fixture.mockFrameRateChangesPerBatch.count * metricsBatches
-        try assertMetricValue(measurements: measurements, key: kSentryProfilerSerializationKeyFrameRates, numberOfReadings: expectedFrameRateReadings, expectedValue: dummyValue, transaction: transaction)
+        try assertMetricEntries(measurements: measurements, key: kSentryProfilerSerializationKeySlowFrameRenders, expectedEntries: fixture.expectedSlowFrames, transaction: transaction)
+        try assertMetricEntries(measurements: measurements, key: kSentryProfilerSerializationKeyFrozenFrameRenders, expectedEntries: fixture.expectedFrozenFrames, transaction: transaction)
+        try assertMetricEntries(measurements: measurements, key: kSentryProfilerSerializationKeyFrameRates, expectedEntries: fixture.expectedFrameRateChanges, transaction: transaction)
 #endif
     }
 
-    func assertMetricValue<T: Equatable>(measurements: [String: Any], key: String, numberOfReadings: Int, expectedValue: T?, transaction: Transaction) throws {
+    func assertMetricEntries(measurements: [String: Any], key: String, expectedEntries: [[String: Any]], transaction: Transaction) throws {
+        guard let metricContainer = measurements[key] as? [String: Any] else {
+            throw TestError.noMetricsReported
+        }
+        guard let actualEntries = metricContainer["values"] as? [[String: Any]] else {
+            throw TestError.malformedMetricValueEntry
+        }
+        XCTAssertEqual(actualEntries.count, expectedEntries.count, "Wrong number of values under \(key)")
+
+        func sortedByTimestamps(_ entries: [[String: Any]]) -> [[String: Any]] {
+            entries.sorted { a, b in
+                UInt64(a["elapsed_since_start_ns"] as! String)! < UInt64(b["elapsed_since_start_ns"] as! String)!
+            }
+        }
+
+        let sortedActualEntries = sortedByTimestamps(actualEntries)
+        let sortedExpectedEntries = sortedByTimestamps(expectedEntries)
+
+        // debug
+        func printTimestamps(entries: [[String: Any]]) {
+            print(entries.reduce(into: [NSString](), { partialResult, entry in
+                partialResult.append(entry["elapsed_since_start_ns"] as! NSString)
+            }))
+        }
+        print("expected entry absolute timestamps for \(key)")
+        printTimestamps(entries: sortedExpectedEntries)
+        print("actual entry relative timestamps for \(key)")
+        printTimestamps(entries: sortedActualEntries)
+        print("transaction start time: \(transaction.startSystemTime)")
+        // /debug
```

Comment:
`m`: I think this is going to spam the test logs. It would be awesome to only print this as an assertion message when the test fails.

Reply:
i like that idea. i was planning to just remove it but it would be helpful